libr/bin/format/mach0/mach0.c

```diff
@@ -3311,92 +3321,89 @@ RSkipList *MACH0_(get_relocs)(struct MACH0_(obj_t) *bin) {
 										break;
 									}
 									*reloc = *ref;
-									reloc->addr = cursor;
+									reloc->addr = addr;
 									reloc->offset = paddr;
 									if (addend != -1) {
 										reloc->addend = addend;
 									}
 									r_skiplist_insert (relocs, reloc);
 								}
-								cursor += delta * wordsize;
+								addr += delta * wordsize;
 								if (!delta) {
 									break;
 								}
 							}
-						} else {
-							bprintf ("Error: BIND_SUBOPCODE_THREADED_APPLY with no current segment\n");
 						}
+						break;
+					default:
+						bprintf ("Error: Unexpected BIND_OPCODE_THREADED sub-opcode: 0x%x\n", imm);
 					}
 					break;
-				default:
-					bprintf ("Error: Unexpected BIND_OPCODE_THREADED sub-opcode: 0x%x\n", imm);
-				}
-				break;
-			}
-			case BIND_OPCODE_SET_DYLIB_ORDINAL_IMM:
-				lib_ord = imm;
-				break;
-			case BIND_OPCODE_SET_DYLIB_ORDINAL_ULEB:
-				lib_ord = ULEB();
-				break;
-			case BIND_OPCODE_SET_DYLIB_SPECIAL_IMM:
-				lib_ord = imm? (st8)(BIND_OPCODE_MASK | imm) : 0;
-				break;
-			case BIND_OPCODE_SET_SYMBOL_TRAILING_FLAGS_IMM: {
-				sym_name = (char*)ur.p;
-				while (*ur.p++ && ur.p<end) {
-					/* empty loop */
 				}
-				if (threaded_binds) {
+				case BIND_OPCODE_SET_DYLIB_ORDINAL_IMM:
+					lib_ord = imm;
 					break;
-				}
-				sym_ord = -1;
-				if (bin->symtab && bin->dysymtab.nundefsym < 0xffff) {
-					for (j = 0; j < bin->dysymtab.nundefsym; j++) {
-						int stridx = 0;
-						int iundefsym = bin->dysymtab.iundefsym;
-						if (iundefsym >= 0 && iundefsym < bin->nsymtab) {
-							int sidx = iundefsym + j;
-							if (sidx < 0 || sidx >= bin->nsymtab) {
-								continue;
+				case BIND_OPCODE_SET_DYLIB_ORDINAL_ULEB:
+					lib_ord = ULEB();
+					break;
+				case BIND_OPCODE_SET_DYLIB_SPECIAL_IMM:
+					lib_ord = imm? (st8)(BIND_OPCODE_MASK | imm) : 0;
+					break;
+				case BIND_OPCODE_SET_SYMBOL_TRAILING_FLAGS_IMM: {
+					sym_name = (char*)ur.p;
+					while (*ur.p++ && ur.p<end) {
+						/* empty loop */
+					}
+					if (threaded_binds) {
+						break;
+					}
+					sym_ord = -1;
+					if (bin->symtab && bin->dysymtab.nundefsym < 0xffff) {
+						for (j = 0; j < bin->dysymtab.nundefsym; j++) {
+							int stridx = 0;
```

Comment:
maybe initialize with -1 in the case the symbol is not found it will be using the first one (index 0), which is imho wrong